### PR TITLE
Add support for LMS to library

### PR DESCRIPTION
**Pull request by abkarcher**
UTs will come at a later date.

App handler includes comment describing outputs.

### CODE REVIEW

**Review by ashman-p — APPROVED** [review]

File: include/acvp/acvp.h [hunk 1]
```diff
@@ -3951,7 +4053,64 @@ ACVP_RESULT acvp_cap_safe_primes_set_parm(ACVP_CTX *ctx,
...
+ *
+ *        This function may be called multiple times to specify more than one crypto parameter
+ *        value for the LMS algorithm. The ACVP_CIPHER value passed to this function should
+ *        already have been setup by invoking acvp_cap_lms_enable().
+ *
+ * @param ctx Pointer to ACVP_CTX that was previously created by calling acvp_create_test_session.
+ * @param cipher ACVP_CIPHER enum value identifying the crypto capability.
+ * @param param ACVP_LMS_PARAM enum value identifying the algorithm parameter that is being
+ *        specified.
+ * @param mode the value corresponding to the parameter being set, at present only generation mode
```
Comment (malbert1) [inline include/acvp/acvp.h:4090]: is the param 'value' ?

File: src/acvp.c [hunk 2]
```diff
@@ -583,6 +586,19 @@ static void acvp_cap_free_kts_ifc_schemes(ACVP_CAPS_LIST *cap_entry) {
     }
     free(cap_entry->cap.kts_ifc_cap->schemes);
 }
+
+static void acvp_cap_free_lms(ACVP_LMS_CAP *cap) {
+    ACVP_LMS_SPECIFIC_LIST *prev = NULL, *curr = cap->specific_list;
```
Comment (malbert1) [inline src/acvp.c:591]: need to check cap <> null or will that always be true ?

File: src/acvp_build_register.c [hunk 3]
```diff
@@ -4535,6 +4538,113 @@ static ACVP_RESULT acvp_build_safe_primes_register_cap(ACVP_CTX *ctx,
     return ACVP_SUCCESS;
 }
 
+static ACVP_RESULT acvp_build_lms_register_cap(ACVP_CTX *ctx,
+                                                       JSON_Object *cap_obj,
+                                                       ACVP_CAPS_LIST *cap_entry) {
```
Comment (malbert1) [inline src/acvp_build_register.c:4543]: formatting seems odd - tabs and spaces, or just the way git is showing it?

File: src/acvp_build_register.c [hunk 4]
```diff
@@ -4535,6 +4538,113 @@ static ACVP_RESULT acvp_build_safe_primes_register_cap(ACVP_CTX *ctx,
...
+    }
+    json_object_set_string(cap_obj, "mode", mode);
+
+    revision = acvp_lookup_cipher_revision(cap_entry->cipher);
+    if (revision == NULL) return ACVP_INVALID_ARG;
+    json_object_set_string(cap_obj, "revision", revision);
+
+    if (cap_entry->prereq_vals) {
+        result = acvp_lookup_prereqVals(cap_obj, cap_entry);
+        if (result != ACVP_SUCCESS) { return result; }
```
Comment (malbert1) [inline src/acvp_build_register.c:4571]: Please use { ... } consistently (see line 4566)

File: src/acvp_util.c [hunk 5]
```diff
@@ -631,6 +631,94 @@ ACVP_CIPHER acvp_lookup_aux_function_alg_tbl(const char *str) {
...
+    { ACVP_LMS_MODE_SHAKE_M32_H20, "LMS_SHAKE_M32_H20"},
+    { ACVP_LMS_MODE_SHAKE_M32_H25, "LMS_SHAKE_M32_H25"}
+};
+
+static int lms_mode_tbl_len = sizeof(lms_mode_tbl) / sizeof(struct acvp_enum_string_pair);
+
+ACVP_LMS_MODE acvp_lookup_lms_mode(const char *str) {
+    int diff = 1, i = 0;
+    for (i = 0; i < lms_mode_tbl_len; i++) {
+    strcmp_s(lms_mode_tbl[i].string, strnlen_s(lms_mode_tbl[i].string, ACVP_ALG_NAME_MAX), str, &diff);
```
Comment (malbert1) [inline src/acvp_util.c:662]: formatting? (indent strcmp_s() )

File: src/acvp_util.c [hunk 6]
```diff
@@ -631,6 +631,94 @@ ACVP_CIPHER acvp_lookup_aux_function_alg_tbl(const char *str) {
...
+    { ACVP_LMOTS_MODE_SHAKE_N32_W4, "LMOTS_SHAKE_N32_W4" },
+    { ACVP_LMOTS_MODE_SHAKE_N32_W8, "LMOTS_SHAKE_N32_W8" }
+};
+
+static int lmots_mode_tbl_len = sizeof(lmots_mode_tbl) / sizeof(struct acvp_enum_string_pair);
+
+ACVP_LMOTS_MODE acvp_lookup_lmots_mode(const char *str) {
+    int diff = 1, i = 0;
+    for (i = 0; i < lmots_mode_tbl_len; i++) {
+    strcmp_s(lmots_mode_tbl[i].string, strnlen_s(lmots_mode_tbl[i].string, ACVP_ALG_NAME_MAX), str, &diff);
```
Comment (malbert1) [inline src/acvp_util.c:704]: formatting

File: include/acvp/acvp.h (hunk 1 above)
Comment (abkarcher) [reply include/acvp/acvp.h]: Fixed, thanks!

File: src/acvp.c (hunk 2 above)
Comment (abkarcher) [reply src/acvp.c:591]: It is called from a union so it won't be null

File: src/acvp_build_register.c (hunk 4 above)
Comment (abkarcher) [reply src/acvp_build_register.c:4571]: Fixed here, thanks

File: src/acvp_build_register.c (hunk 3 above)
Comment (abkarcher) [reply src/acvp_build_register.c]: All spaces, just too many in a few places somehow. fixed

**Review by malbert1 — APPROVED** [review]

**Pull request merged**